vars/transportRequestCreate.groovy

```diff
@@ -0,0 +1,68 @@
+import com.sap.piper.GitUtils
+import groovy.transform.Field
+
+import com.sap.piper.ConfigurationMerger
+import com.sap.piper.cm.ChangeManagement
+import com.sap.piper.cm.ChangeManagementException
+
+import hudson.AbortException
+
+
+@Field def STEP_NAME = 'transportRequestCreate'
+
+@Field Set parameterKeys = [
+    'changeId',
+    'developmentSystemId',
+    'cmCredentialsId',
+    'cmEndpoint'
+  ]
+
+@Field Set stepConfigurationKeys = [
+    'cmCredentialsId',
+    'cmEndpoint'
+  ]
+
+def call(parameters = [:]) {
+
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
+
+        def script = parameters?.script ?: [commonPipelineEnvironment: commonPipelineEnvironment]
+
+        ChangeManagement cm = new ChangeManagement(script)
+
+        Map configuration = ConfigurationMerger.merge(parameters.script, STEP_NAME,
+                                                      parameters, parameterKeys,
+                                                      stepConfigurationKeys)
+
+        def changeId = configuration.changeId
```

Comment:
ChangeId is expected to be contained in one of the commit messages. Could be retrieved from the commit message(s). Would be in general possible to implement a step retrieving the commit message and to put it into the commonPipelineEnvironment, but why having an extra auxiliar step for this. Can be done inside each of the cm steps. Of course we can think about caching it in the common pipeline environment, but since we cannot rely on the order of the cm related steps each step should try to resolve it from the changes in case it is not contained in the common pipeline environment.

It is OK to provide the changeId also via signature, but IMO we should also have the other approach as outlined above. I believe the normals way is to retrieve it from the commits inside the step.

Reply:
I think that the changeId should be retrieve on a previous step, and it should be passed as parameter to this step.

Reply:
As already written above we should keep the changeId in the signature, there is nothing wrong with that. I doubt that this will be used frequently in a production environment, but it might be helpful for troubleshooting.

Overall I suggest the following approach:

1.) In case the changeId if provided via signature we take that value
2.) (optional (but reasonable)) caching: we retrieve the changeId from the commonPIpelineEnvironment.
3.) The step retrieves the changeId from the git commits (optional: and puts it into the commonPipeline environment so that it can be retrieved from there by steps later in the pipeline.)

Reply:
Ok, yes, you are right, but can I do it in another PR, once #162 and #166 are merged?

Reply:
I'm already working on this. See corresponding commit on branch 'cm' in my fork.